packages/arb-token-bridge-ui/src/token-bridge-sdk/OftV2TransferStarter.ts

```diff
@@ -158,9 +196,26 @@ export class OftV2TransferStarter extends BridgeTransferStarter {
     return contract.functions.approve(spender, constants.MaxUint256) // Eth USDT will need MAX approval since that cannot be changed afterwards
   }
 
-  // for OFT, we don't have functions for gas estimates, `sendQuote` method tells us the fees directly
-  public async transferEstimateGas() {
-    return undefined
+  public async transferEstimateGas({
+    amount,
+    signer,
+    destinationAddress
+  }: TransferEstimateGasProps) {
+    await this.validateOftTransfer()
+
+    const oftContract = this.getOftAdapterContract(signer)
+    const config = await prepareTransferConfig({
+      signer,
+      oftContract,
+      amount,
+      destLzEndpointId: this.destLzEndpointId!,
+      destinationAddress
+    })
+
+    return {
+      estimatedParentChainGas: await signer.estimateGas(config),
+      estimatedChildChainGas: constants.Zero
+    }
   }
 
   public async transferEstimateFee({
```

Comment:
I need to add the logic to subtract the gas from the fee so we can have an estimate of only the fee from layerzero